TELCODOCS-756-AWS: First draft (#53022)
openshift/openshift-docs: modules/nvidia-gpu-aws-deploying-the-node-feature-discovery-operator.adoc

// Module included in the following assemblies:
//

> //  * installing/installing-nvidia-gpu-aws.adoc

Reviewer: Update this one too :)

:_content-type: PROCEDURE
[id="nvidia-gpu-aws-deploying-the-node-feature-discovery-operator_{context}"]
= Deploying the Node Feature Discovery Operator

> After the GPU-enabled node is created, you need to discover the GPU enabled node

Reviewer: ```suggestion
After the GPU-enabled node is created, you need to discover the GPU-enabled node
```

so it can be scheduled. To do this, install the Node Feature Discovery (NFD) Operator.

> The NFD operator identifies hardware device features in nodes.

Reviewer: There are a few instances of `operator`, they all should be `Operator`

```suggestion
The NFD Operator identifies hardware device features in nodes.
```

It solves the general problem of identifying and cataloging hardware resources in

> the infrastructure nodes so they can be made available to OpenShift.

Reviewer: ```suggestion
the infrastructure nodes so they can be made available to {product-title}.
```

.Procedure

. Install the Node Feature Discovery operator from **OperatorHub** in the

> **OpenShift Container Platform** console.

Reviewer: ```suggestion
{product-title} console.
```

> . After installing the NFD operator into **OperatorHub**,select **Node Feature Discovery** from the installed operators list and select **Create instance**.

Reviewer: ```suggestion
. After installing the NFD operator into **OperatorHub**, select **Node Feature Discovery** from the installed Operators list and select **Create instance**.
```

> This installs the ``nfd-master`` and ``nfd-worker`` pods, one ``nfd-worker`` pod for each worker node, in the ``openshift-nfd`` namespace.

Reviewer: ```suggestion
This installs the ``nfd-master`` and ``nfd-worker`` pods, one ``nfd-worker`` pod for each compute node, in the ``openshift-nfd`` namespace.
```

> . Verify the Operator is installed and running:

Reviewer: ```suggestion
. Verify that the Operator is installed and running by running the following command:
```

+
[source,terminal]
----
$ oc get pods -n openshift-nfd
----
+
.Example output
+
[source,terminal]
----
NAME                                       READY    STATUS     RESTARTS   AGE

nfd-controller-manager-8646fcbb65-x5qgk    2/2      Running 7  (8h ago)   1d
----

. Browse to the installed operator in the console and select
**Create Node Feature Discovery**.

> . Select **Create** to build a NFD Custom Resource. This creates NFD pods in

Reviewer: ```suggestion
. Select **Create** to build a NFD custom resource. This creates NFD pods in
```

> the ``openshift-nfd`` namespace that poll the OpenShift nodes for hardware

Reviewer: ```suggestion
the ``openshift-nfd`` namespace that poll the {product-title} nodes for hardware
```

resources and catalogue them.

> . After a successful build, verify that a NFD pod is running on each nodes:

Reviewer: ```suggestion
. After a successful build, verify that a NFD pod is running on each nodes by running the following command:
```

+
[source,terminal]
----
$ oc get pods -n openshift-nfd
----
+
.Example output
[source,terminal]
----
NAME                                       READY   STATUS      RESTARTS        AGE
nfd-controller-manager-8646fcbb65-x5qgk    2/2     Running     7 (8h ago)      12d
nfd-master-769656c4cb-w9vrv                1/1     Running     0               12d
nfd-worker-qjxb2                           1/1     Running     3 (3d14h ago)   12d
nfd-worker-xtz9b                           1/1     Running     5 (3d14h ago)   12d
----
+
The NFD operator uses vendor PCI IDs to identify hardware in a node.

> NVIDIA uses the PCI ID 10de.

Reviewer: ```suggestion
NVIDIA uses the PCI ID `10de`.
```

> . View the NVIDIA GPU discovered by the NFD operator:

Reviewer: ```suggestion
. View the NVIDIA GPU discovered by the NFD operator by running the following command:
```

+
[source,terminal]
----
$ oc describe node ip-10-0-132-138.us-east-2.compute.internal | egrep 'Roles|pci'
----
+
.Example output
[source,terminal]
----
Roles: worker

feature.node.kubernetes.io/pci-1013.present=true

feature.node.kubernetes.io/pci-10de.present=true

feature.node.kubernetes.io/pci-1d0f.present=true
----
+
``10de`` appears in the node feature list for the GPU-enabled node.

> This mean the NFD operator correctly identified the node from the GPU-enabled MachineSet.

Reviewer: Could move these last two steps under a `.Verification` section

// Module included in the following assemblies:
//
//  * machine_management/creating_machinesets/creating-machineset-aws.adoc

:_content-type: PROCEDURE
[id="nvidia-gpu-aws-deploying-the-node-feature-discovery-operator_{context}"]
= Deploying the Node Feature Discovery Operator

After the GPU-enabled node is created, you need to discover the GPU-enabled node
so it can be scheduled. To do this, install the Node Feature Discovery (NFD) Operator.

The NFD Operator identifies hardware device features in nodes.
It solves the general problem of identifying and cataloging hardware resources in
the infrastructure nodes so they can be made available to {product-title}.

.Procedure

. Install the Node Feature Discovery Operator from *OperatorHub* in the {product-title} console.

> . After installing the NFD Operator into *OperatorHub*,select *Node Feature Discovery* from the

Reviewer: ```suggestion
. After installing the NFD Operator into *OperatorHub*, select *Node Feature Discovery* from the
```

// Module included in the following assemblies:
//
//  * installing/installing-nvidia-gpu-aws.adoc

:_content-type: PROCEDURE
[id="nvidia-gpu-aws-deploying-the-node-feature-discovery-operator_{context}"]
= Deploying the Node Feature Discovery Operator

After the GPU-enabled node is created, you need to discover the GPU enabled node
so it can be scheduled. To do this, install the Node Feature Discovery (NFD) Operator.

The NFD operator identifies hardware device features in nodes.
It solves the general problem of identifying and cataloging hardware resources in
the infrastructure nodes so they can be made available to OpenShift.

.Procedure

. Install the Node Feature Discovery operator from **OperatorHub** in the
**OpenShift Container Platform** console.

. After installing the NFD operator into **OperatorHub**,
select **Node Feature Discovery** from the installed operators list and
select **Create instance**. This installs the ``openshift-nfd`` Operator

> into the ``openshift-operators`` namespace.

Reviewer 2: The sentence is very outdated.  it should read:  This installs the "nfd-master" and "nfd-worker" pods (one nfd-worker pod for each worker node) in the "openshift-nfd" namespace.

. Verify the Operator is installed and running:
+
[source,terminal]
----

> $ oc get pods -n openshift-operators

Reviewer 2: The namespace for NFD operator is outdated.  It should read:
$ oc get pods -n openshift-nfd

Author: Made this change.

----
+
.Example output
+
[source,terminal]
----
NAME READY STATUS RESTARTS AGE

> nfd-operator-fd55688bd-4rrkq 1/1 Running 0 18m

Reviewer 2: `nfd-controller-manager-8646fcbb65-x5qgk                           2/2     Running     7 (8h ago)      1d`

Author: Made this change.

----

. Browse to the installed operator in the console and select
**Create Node Feature Discovery**.

. Select **Create** to build a NFD Custom Resource. This creates NFD pods in
the ``openshift-nfd`` namespace that poll the OpenShift nodes for hardware
resources and catalogue them.

. After a successful build, verify that a NFD pod is running on each nodes:
+
[source,terminal]
----
$ oc get pods -n openshift-nfd
----
+
.Example output
[source,terminal]
----
NAME READY STATUS RESTARTS AGE

nfd-master-mc99f 1/1 Running 0 51s

nfd-master-t7rrl 1/1 Running 0 51s

nfd-master-w9pgx 1/1 Running 0 51s

> 

Reviewer 2: Starting in OCP 4.12, there will be one nfd-master deployment pod per cluster, instead of one for each master node.

Author: @wabouhamad Could you provide a sample?

Reviewer 2: @StephenJamesSmith : here's an example from OCP 4.12
`# oc get pods -n openshift-nfd
NAME                                                              READY   STATUS      RESTARTS        AGE
nfd-controller-manager-8646fcbb65-x5qgk                           2/2     Running     7 (8h ago)      12d
nfd-master-769656c4cb-w9vrv                                       1/1     Running     0               12d
nfd-worker-qjxb2                                                  1/1     Running     3 (3d14h ago)   12d
nfd-worker-xtz9b                                                  1/1     Running     5 (3d14h ago)   12d`

Author: Ok. Thanks. Made this change.
